src/binding.cc

```diff
@@ -26,89 +27,72 @@
 #define ICONV_CONST
 #endif  // ICONV_CONST
 
-namespace
-{
-
-using v8::Array;
-using v8::Boolean;
-using v8::FunctionTemplate;
-using v8::Handle;
-using v8::Integer;
-using v8::Local;
-using v8::Null;
-using v8::Object;
-using v8::ObjectTemplate;
-using v8::String;
-using v8::Value;
 
+using namespace Napi;
 
-struct Iconv
+struct Iconv: Napi::ObjectWrap<Iconv>
 {
-  static Nan::Persistent<ObjectTemplate> object_template;
-  iconv_t conv_;
+  static Napi::FunctionReference object_template; // iconv_constructor
 
-  Iconv(iconv_t conv)
-  {
-    conv_ = conv;
-  }
+  Iconv(const Napi::CallbackInfo& info): 
+    Napi::ObjectWrap<Iconv>(info) {
+      conv_ = info[0].As<Napi::External<iconv_t>>().Data();
+    }
+  iconv_t conv_;
 
   ~Iconv()
   {
     iconv_close(conv_);
   }
 
-  static void WeakCallback(const Nan::WeakCallbackInfo<Iconv>& data)
-  {
-    delete data.GetParameter();
+  static void Init(Napi::Env env) {
+    Napi::Function func = DefineClass(env, "Iconv", {});
+    Iconv::object_template = Napi::Persistent(func);
+    Iconv::object_template.SuppressDestruct();
   }
 
-  static void Initialize(Handle<Object> obj)
+  static Napi::Object Initialize(Napi::Env env, Napi::Object exports)
   {
-    Local<ObjectTemplate> t = Nan::New<ObjectTemplate>();
-    t->SetInternalFieldCount(1);
-    object_template.Reset(t);
-    obj->Set(Nan::New<String>("make").ToLocalChecked(),
-             Nan::New<FunctionTemplate>(Make)->GetFunction());
-    obj->Set(Nan::New<String>("convert").ToLocalChecked(),
-             Nan::New<FunctionTemplate>(Convert)->GetFunction());
+    Iconv::Init(env);
+    exports.Set(Napi::String::New(env, "make"),
+             Napi::Function::New(env, Make));
```

Comment:
This line should be indented so the `Napi::...` is in the same column as the line above.